docs/reference/api/platform/Error.md

```diff
@@ -1,5 +1,21 @@
 ## Error
 
-The error function generates a fatal runtime error, which allows the user to dump a message to the serial line, halt the program and place the device in an idle mode with a blinking red LED.
+Mbed OS provides an error function to output messages to `STDIO` at runtime when the system encounters a fatal error and causes the application to terminate. `mbed_error.h` declares this error message. Note that the error function outputs the error message in debug and develop builds only. In release builds, the error function does not generate any output, but the application is still be terminated.
```

Comment:
Query: Who or what terminates the application? Please clarify for active voice. [Note: I'll fix that extra "be" that I accidentally left in there after I know how to change the passive to active voice.]

Reply:
Good question, its the mbed-OS error function implementation that terminates the application.